src/student/student.service.ts

```diff
@@ -0,0 +1,120 @@
+            expectedContractType: oneStudent.expectedContractType,
+            expectedSalary: oneStudent.expectedSalary,
+            expectedTypeWork: oneStudent.expectedTypeWork,
+            firstName: oneStudent.firstName,
+            githubUsername: oneStudent.githubUsername,
+            lastName: oneStudent.lastName,
+            monthsOfCommercialExp: oneStudent.monthsOfCommercialExp,
+            portfolioUrls: oneStudent.portfolioUrls,
+            projectUrls: oneStudent.projectUrls,
+            targetWorkCity: oneStudent.targetWorkCity,
+            telephone: oneStudent.telephone,
+            workExperience: oneStudent.workExperience
+        };
+
+    }
+
+    async getOneStudentAndCompareToken(id: string, registerToken: string) {
+        const oneStudent = await this.dataSource
+            .getRepository(Student)
+            .createQueryBuilder('student')
+            .leftJoinAndSelect('student.studentImport', 'studentImport')
+            .where('studentImport.id = :id', {id})
+            .andWhere('studentImport.registerToken = :registerToken', {registerToken})
+            .getOne()
+
+        return oneStudent ? {
+            isOk: true,
+            id: oneStudent.studentImport.id,
+            registerToken: oneStudent.studentImport.registerToken
+        } : {
+            isOk: false,
+        }
+    }
+
+    async createStudent(id: string, createStudentDto: CreateStudentDto) {
+        const student = new Student();
+        student.studentImport = await StudentImport.findOneBy({
+            id,
+        })
+        student.id = uuidv4();
+        student.pwdHash = hashPwd(createStudentDto.pwdHash);
+        student.expectedContractType = createStudentDto.expectedContractType;
+        student.expectedTypeWork = createStudentDto.expectedTypeWork;
+        student.bio = createStudentDto.bio;
+        student.courses = createStudentDto.courses;
+        student.canTakeApprenticeship = createStudentDto.canTakeApprenticeship;
+        student.education = createStudentDto.education;
+        student.expectedSalary = createStudentDto.expectedSalary;
+        student.projectUrls = createStudentDto.projectUrls;
+        student.firstName = createStudentDto.firstName;
+        student.githubUsername = createStudentDto.githubUsername;
+        student.lastName = createStudentDto.lastName;
+        student.monthsOfCommercialExp = createStudentDto.monthsOfCommercialExp;
+        student.portfolioUrls = createStudentDto.portfolioUrls;
+        student.targetWorkCity = createStudentDto.targetWorkCity;
+        student.telephone = createStudentDto.telephone;
+        student.workExperience = createStudentDto.workExperience;
+        await student.save()
+    }
+
+    async patchStudent(id: string, updateStudentDto: UpdateStudentDto) {
+        await this.dataSource
+            .createQueryBuilder()
+            .update(Student)
+            .set({
+                pwdHash: hashPwd(updateStudentDto.pwdHash),
```

Comment:
Setting new password will be in different endpoint. Here has to be deleted.

Reply:
Adam, I imagined register form as a huge form with password too. Can You explain us Your vision on discord? :D